src/frontend/js/components/ReferralForm/TopicField.tsx

```diff
@@ -158,16 +171,45 @@ export const TopicField: React.FC<TopicFieldProps> = ({
           <TopicSuggestion {...{ topic, isHighlighted }} />
         )}
         shouldRenderSuggestions={() => true}
+        renderSuggestionsContainer={({ containerProps, children, query }) => (
+          <div {...containerProps}>
+            {children}
+
+            {!children &&
+              query.length > 0 &&
+              state.context.value.length === 0 &&
+              isInputFocused && (
+                <div>
+                  <FormattedMessage {...messages.noItemsfound} />
+                </div>
+              )}
+          </div>
+        )}
         inputProps={{
           id: seed('referral-topic-label'),
           'aria-describedby': seed('referral-topic-description'),
           placeholder: intl.formatMessage(messages.label),
           onChange: (_, { newValue }) => {
+            if (state.context.value.length > 0)
+              send({ type: 'CHANGE', data: '' });
+
             setValue(newValue);
           },
+          onFocus: () => {
+            setisInputFocused(true);
+          },
+          onBlur: () => {
+            send('CLEAN');
+            setisInputFocused(false);
+          },
           value,
         }}
       />
+      {state.matches('cleaned.true') && state.matches('validation.invalid') ? (
+        <div className="mt-4 text-danger-600">
```

Comment:
This is good, but I think we can make it even better by showing a different message when there is nothing in the box, like "This field is mandatory. Please select a topic in the list.", which is clearer when the field is left empty on blur.